src/app/content/components/Page/highlightManager.ts

```diff
@@ -49,8 +54,12 @@ const onSelectHighlight = (
   services: Services,
   highlights: Highlight[],
   highlight: Highlight | undefined
-) => defer(() => {
-  if (highlights.length > 0 || !highlight || services.getProp().focused) {
+) => defer(async() => {
+  if (highlights.length > 0 || !highlight) {
+    return;
+  }
+
+  if (services.getProp().focused && services.getProp().hasUnsavedHighlight && !await showConfirmation()) {
```

Comment:
do you really need the `services.getProp().focused && ` part of this? does it not get cleared when focus is lost?